### PR TITLE
Enable query string filtering in existing views

### PR DESCRIPTION
**Pull request by CDimonaco**
# Description

This PR enables the query string filter capability on every filtered table column in the application.

This enable the users to share their links with filters in order to have a more consistent UX across the tables.

Feature enabled in

- Hosts List
- Clusters List
- Sap Systems List
- Databases List

The gif will show the filtering feature in every screen.

trento_filters_all_tables

Tag @jagabomb for UX review.

## How was this tested?

The table filtering feature was already test with e2e test in another PR.

### CODE REVIEW

**Comment by abravosuse** [comment]
Code apart, it looks good to me.

**Review by dottorblaster — CHANGES_REQUESTED** [review]
LGTM but could we also cover some of this with Jest? Consider using `renderWithRouter` and `withState` helpers :+1:

**Comment by CDimonaco** [comment]
> LGTM but could we also cover some of this with Jest? Consider using `renderWithRouter` and `withState` helpers +1

I think we could test the table component with jest because we don't have any dependency to the router, but I will investigate on that stuff

**Review by jagabomb — APPROVED** [review]
Tested this manually and it seems to be updating the url correctly across the relevant pages. Links also seem to work when loaded in a new browser window with a pre-authenticated session.

**Review by rtorrero — APPROVED** [review]
Green light from my side but if we can get those tests it'd be great :-)

File: assets/js/lib/test-utils/index.jsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { setSapSystems } from '@state/sapSystems';
 
 import hosts from './data/hosts';
 import clusters from './data/clusters';
-import sapSystems from './data/hosts';
+import sapSystems from './data/sapSystems';
```
Comment (dottorblaster) [inline assets/js/lib/test-utils/index.jsx:13]: Ouch ._. Sorry about that

**Review by dottorblaster — APPROVED** [review]
Aw yeeeeeah

**Comment by CDimonaco** [comment]
Add jest tests for every view except the database view, but we already tested the functionality in the tables. I skipped the database view testing for some problems in the fixtures (I need to connect some dots 👁️ ) so maybe we could track in another pr @dottorblaster what do you think?

**Pull request merged**